### PR TITLE
[CARBONDATA-714]Documented how to handle bad records

### PR DESCRIPTION
**Pull request by PallaviSingh1992**

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/973/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/983/

**Comment by PallaviSingh1992** [comment]
@chenliang613 please review

File: docs/faq.md [hunk 1]
```diff
@@ -18,30 +18,57 @@
...
+Try creating ``carbonsession`` with ``storepath`` specified in the following manner :
+```
+val carbon = SparkSession.builder().config(sc.getConf).getOrCreateCarbonSession(<store_path>)
+```
+Example:
+```
+val carbon = SparkSession.builder().config(sc.getConf).getOrCreateCarbonSession("hdfs://localhost:9000/carbon/store ")
+```
+
+## What is Carbon Lock Type?
```
Comment (chenliang613) [inline docs/faq.md]: For users, which scenario need to set this parameter for lock?

File: docs/faq.md [hunk 2]
```diff
@@ -18,30 +18,57 @@
...
+```
+'BAD_RECORDS_ACTION'='REDIRECT'
+```
+
+- To ignore the Bad Records from getting stored in the raw csv, we need to set the following in the query :
+```
+'BAD_RECORDS_ACTION'='INDIRECT'
+```
+
+## How to resolve store location can not be found?
```
Comment (chenliang613) [inline docs/faq.md]: Seems the title should be : How to specify storelocation while creating carbonsession

File: docs/faq.md [hunk 3]
```diff
@@ -18,30 +18,57 @@
...
+## What are Bad Records?
+Records that fail to get loaded into the CarbonData due to data type incompatibility or are empty or have incompatible format are classified as Bad Records.
 
-    You need to specify the spark version while using Maven to build project.
+## Where are Bad Records Stored in CarbonData?
+The bad records are stored at the location set in carbon.badRecords.location in carbon.properties file.
+By default **carbon.badRecords.location** specifies the following location ``/opt/Carbon/Spark/badrecords``.
 
-* **Getting NotImplementedException for subquery using IN and EXISTS**
+## How to handle Bad Records?
```
Comment (chenliang613) [inline docs/faq.md]: This is "how to enable bad record logging".

File: docs/faq.md [hunk 4]
```diff
@@ -18,30 +18,57 @@
...
-    
-* **Getting Exceptions on creating  a view**
-    
-    View not supported in CarbonData.
-    
-* **How to verify if ColumnGroups have been created as desired.**
+- To pad the incorrect values of the csv rows with NULL value and load the data in CarbonData, set the following in the query :
+```
+'BAD_RECORDS_ACTION'='FORCE'
+```
```
Comment (chenliang613) [inline docs/faq.md]: Please add "How to ignore the bad records" ?
Please find the detail discussion at here : http://apache-carbondata-mailing-list-archive.1130556.n5.nabble.com/data-lost-when-loading-data-from-csv-file-to-carbon-table-td7554.html

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**